### PR TITLE
fix(purchase): throw error on invalid item account

### LINKED ISSUE #2901: Prevent Billing and Ordering
- Prevent billing of items that are missing profit accounts
- Prevent the ordering of items that are missing expense and inventory accounts

### PR DESCRIPTION
**Pull request by jniles**
This commit blocks invalid purchase items when their accounts are not correctly set.

Closes #2901.

finalpurchaseerrorfeature
_Fig 1: Error message on purchase_

### CODE REVIEW

**Comment by jniles** [comment]
@lomamech, this is finally ready for review.

File: client/src/modules/purchases/create/PurchaseOrderForm.js [hunk 1]
```diff
@@ -93,13 +93,16 @@ function PurchaseOrderFormService(Inventory, AppCache, Store, Pool, PurchaseOrde
     this.digest();
 
     // filters out valid items
-    const invalidItems = this.store.data.filter((row) => {
-      return row._invalid;
-    });
+    const invalidItems = this.store.data.filter(row => row._invalid);
```
Comment (mbayopanda) [inline client/src/modules/purchases/create/PurchaseOrderForm.js:96]: :+1

File: client/src/modules/purchases/create/PurchaseOrderItemService.js [hunk 2]
```diff
@@ -85,6 +84,9 @@ function PurchaseOrderItemService(uuid) {
     this.inventory_uuid = inventoryItem.uuid;
     this.unit = inventoryItem.unit;
 
+    // make sure the appropriate accounts are defined for inventory items
+    this._hasValidAccounts = inventoryItem.stock_account && inventoryItem.cogs_account;
```
Comment (mbayopanda) [inline client/src/modules/purchases/create/PurchaseOrderItemService.js:88]: 👍

**Review by mbayopanda — APPROVED** [review]
LGTM

bors r+

**Pull request merged**